### PR TITLE
Use pkgdef files to control MIEngine logging

### PR DESCRIPTION
**Pull request by gregg-miskelly**
This checkin changes SetMIDebugLogging.cmd to use pkgdef files instead of directly writing to the registry. This should resolve issues that customers who have multiple _Config keys have been having. I believe this will also fix logging for Dev15, though I don't have a Dev15 machine with the Android tools installed at the moment to verify.

### CODE REVIEW

**Comment by gregg-miskelly** [comment]
@rajkumar42 can you review?
cc @faxue-msft

File: src/MICore/SetMIDebugLogging.cmd [hunk 1]
```diff
@@ -42,68 +42,83 @@ if NOT "%~1"=="" goto :ArgLoop
...
-goto eof
+set SetLoggingError=
+if NOT "%VSRootDir%"=="" call :SetLogging "%VSRootDir%" & goto Done
+
+set ProgRoot=%ProgramFiles(x86)%
+if "%ProgRoot%"=="" set ProgRoot=%ProgramFiles%
+
+set VSVersionFound=
+set MIEngineFound=
+call :TryVSPath "%ProgRoot%\Microsoft Visual Studio 14.0"
```
Comment (rajkumar42) [inline src/MICore/SetMIDebugLogging.cmd]: Should it be 
call :TryVSPath %VSRootDir%

File: src/MICore/SetMIDebugLogging.cmd [hunk 2]
```diff
@@ -42,68 +42,83 @@ if NOT "%~1"=="" goto :ArgLoop
...
+call :TryVSPath "%ProgRoot%\Microsoft Visual Studio\VS15Preview"
```
Comment (rajkumar42) [inline src/MICore/SetMIDebugLogging.cmd]: Do we want to enable/disable logging for Dev14 & Dev15 at the same time. (I'm fine with that).

**Comment by rajkumar42** [comment]
:shipit:

**Pull request merged**